DuckDuckGo/CrashCollectionOnboarding.swift

```diff
@@ -35,16 +36,34 @@ final class CrashCollectionOnboardingViewController: UIHostingController<CrashCo
 }
 
 final class CrashCollectionOnboarding: NSObject {
+    
+    private var featureFlagger: FeatureFlagger
 
-    init(appSettings: AppSettings) {
+    init(appSettings: AppSettings,
+         featureFlagger: FeatureFlagger = AppDependencyProvider.shared.featureFlagger) {
         self.appSettings = appSettings
         self.viewModel = CrashCollectionOnboardingViewModel(appSettings: appSettings)
+        self.featureFlagger = featureFlagger
         super.init()
     }
+    
+    // If the user's crash report opt in status should be reset to unknown upon next release,
+    // increment this value by 1
+    private let crashCollectionShouldRevertOptedInStatusTriggerTargetValue: Int = 1
 
     @MainActor
     func presentOnboardingIfNeeded(for payloads: [Data], from viewController: UIViewController, sendReport: @escaping () -> Void) {
         let isCurrentlyPresenting = viewController.presentedViewController != nil
+        
+        if featureFlagger.isFeatureOn(.crashReportOptInStatusResetting) {
+            if appSettings.crashCollectionOptInStatus == .optedIn &&
+                appSettings.crashCollectionShouldRevertOptedInStatusTrigger < crashCollectionShouldRevertOptedInStatusTriggerTargetValue {
+                debugPrint("Resettting crash report opt in status from \(appSettings.crashCollectionOptInStatus) to .undetermined")
```

Comment:
We try to avoid calls to `print` and `debugPrint` in the code. You could use `Logger.general.debug` instead.